pkg/formats/influx/influx.go

```diff
@@ -534,14 +536,25 @@ func (f *InfluxFormat) fromSnmpDeviceMetric(in *kt.JCHF) []InfluxData {
 				pts := strings.SplitN(metricName, namespaceTokenSep, 2)
 				metricName = pts[1]
 			}
-			if name.Format == kt.FloatMS {
+			switch name.Format {
+			case kt.FloatMS:
 				results = append(results, InfluxData{
 					Name:        f.config.MeasurementPrefix + mib,
 					FieldsFloat: map[string]float64{metricName: float64(float64(in.CustomBigInt[m]) / 1000)},
 					Timestamp:   in.Timestamp,
 					Tags:        attrNew,
 				})
-			} else {
+			case kt.Rate:
+				delta, found := f.getDelta(m, attrNew, in.CustomBigInt[m])
+				if found {
+					results = append(results, InfluxData{
+						Name:      f.config.MeasurementPrefix + mib,
+						Fields:    map[string]int64{metricName: delta},
```

Comment:
Don't you need to divide the delta by the time since the previous poll to convert a counter to a rate? Reporting the counter delta but not the time delta will just force the backend to figure out the time delta. Easier to do it here where that state can be kept in memory.

Reply:
That's a good q!  The doing it at query time way is one approach. For example, NR has a few neat functions like: [func-rate](https://docs.newrelic.com/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions/#func-rate) which lets you mess with things to see both rate and not rate without storing data twice. 

Since there's a uptime ticker here anyway I can just add it in.